corehq/apps/userreports/models.py

```diff
@@ -223,6 +223,12 @@ def all(cls):
             yield cls.wrap(result)
 
 
+class ReportMeta(DocumentSchema):
+    # `True` if this report was initially constructed by the report builder.
+    created_by_builder = BooleanProperty(default=False)
+    builder_report_type = StringProperty(choices=['chart', 'list', 'table', 'worker'])
```

Comment:
`created_by_builder` is used to decide if this report can be edited in the report builder. `builder_report_type` helps decide which form to use when editing the report in the report builder.